cpp/include/raft/core/device_resources_snmg.hpp

```diff
@@ -47,195 +41,100 @@ class device_resources_snmg : public device_resources {
    */
   device_resources_snmg() : device_resources()
   {
-    RAFT_CUDA_TRY(cudaGetDevice(&main_gpu_id_));
+    int main_gpu_id;
+    RAFT_CUDA_TRY(cudaGetDevice(&main_gpu_id));
+    raft::resource::set_main_gpu_id(*this, main_gpu_id);
 
-    // set root rank
-    int root_rank = 0;
-    raft::resource::set_nccl_clique_root_rank(*this, root_rank);
+    raft::resource::set_root_rank(*this, 0);
 
-    // initialize the NCCL clique
-    std::vector<raft::resources>& clique = raft::resource::get_nccl_clique(*this);
-    _init_clique(clique);
-    _init_nccl_comms(clique);
+    // initialize all resources
+    std::vector<raft::resources>& world_resources = raft::resource::get_multi_gpu_resource(*this);
+    _init_world(world_resources);
 
-    RAFT_CUDA_TRY(cudaSetDevice(main_gpu_id_));
+    RAFT_CUDA_TRY(cudaSetDevice(main_gpu_id));
   }
 
   /**
    * @brief Construct a SNMG resources instance with a subset of available GPUs
    *
-   * @param[in] device_ids List of device IDs to be used by the NCCL clique
+   * @param[in] device_ids List of device IDs to be used
    */
   device_resources_snmg(const std::vector<int>& device_ids) : device_resources()
   {
-    RAFT_CUDA_TRY(cudaGetDevice(&main_gpu_id_));
-
-    // set root rank
-    int root_rank = 0;
-    raft::resource::set_nccl_clique_root_rank(*this, root_rank);
-
-    // initialize the NCCL clique
-    std::vector<raft::resources>& clique = raft::resource::get_nccl_clique(*this);
-    _init_clique(clique, device_ids);
-    _init_nccl_comms(clique);
-
-    RAFT_CUDA_TRY(cudaSetDevice(main_gpu_id_));
+    int main_gpu_id;
+    RAFT_CUDA_TRY(cudaGetDevice(&main_gpu_id));
```

Comment:
The main GPU would need to be set in the resource here.

Reply:
thanks for pointing out!